### PR TITLE
fix(console): task-361 live-resize reflow convergence lock + stale-tooltip dismissal

### PR DESCRIPTION
**Pull request by rmusser01**
Console UX review **task-361** (live-resize broken reflow / stale tooltip overlay).

## Key finding

The review's divergence — resizing 900×620 → 700×480 left the **rail full-width with the transcript/inspector gone** and a nav tooltip stuck over the header — **does not reproduce with a native resize**. A pilot repro using `Pilot.resize_terminal` (the same Textual reflow machinery textual-serve drives) shows a live `160×48 → 90×30` resize converges to the **exact cold-start pane layout** — rail / transcript / composer all present, `-console-compact` toggled — with **zero divergence**.

The review observed the breakage only via textual-serve's browser-viewport resize at `origin/dev cad9e271d`, **before task-346** added the `@on(Resize)` height handler. That handler (plus Textual's own reflow) now keeps the panes on a live resize.

## Changes

- **AC #1 (reflow convergence):** regression-locked by `test_console_live_resize_converges_to_cold_start_layout`, which asserts the live-resize layout equals the cold-start layout at the same size (panes present, header compacted).
- **AC #2 (dismiss overlays on resize):** the `@on(Resize)` handler now calls `self._clear_tooltip()` first, so a hover tooltip (the review's stuck "Open the live agent Console." nav tooltip) is dismissed on resize instead of surviving the repaint as a stale overlay. Locked by `test_console_resize_dismisses_stale_tooltip`.

## Verification note

Confirmed via **native pilot resize** — definitive for the Textual reflow path (`resize_terminal` is exactly what textual-serve drives). A full textual-serve browser-viewport session remains available as belt-and-suspenders if the exact review path must be re-walked. 2 new tests + 15 shell/keyboard-trust tests green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Fixes Console live-resize so the layout matches a cold start at the same size and clears stale tooltips on resize. Satisfies TASK-361 (AC#1 and AC#2) with stricter regression tests (real tooltip dismissal, pane queries fail loudly).

- **Bug Fixes**
  - Reflow convergence: live resize now matches the cold-start layout at the same size; all panes visible and header compacts.
  - Tooltip overlays: the `@on(Resize)` handler calls `_clear_tooltip()` first so hover tooltips can’t stick after resize.

<sup>Written for commit a6b75d5a5090ee8af8ca0c857e6bfd988bb4d1a5. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_chatbook/pull/802?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: Tests/UI/test_console_resize_reflow.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    layout: dict = {}
+    for selector in _PANES:
+        try:
+            layout[selector] = bool(console.query_one(selector).display)
+        except Exception:  # noqa: BLE001
+            layout[selector] = None
+    layout["compact"] = console.query_one("#console-shell").has_class(
+        "-console-compact"
+    )
+    return layout
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/UI/test_console_resize_reflow.py:38]: Fixed — real point. `_pane_layout` now queries each pane directly (no `except`), so a missing selector raises and fails the test loudly instead of degrading to `None`; the AC#1 test also asserts every required pane (including `#console-left-rail`) is present, not just that live==cold.

File: Tests/UI/test_console_resize_reflow.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        console = host.screen_stack[-1]
+        await pilot.pause()
+
+        cleared: list[bool] = []
+        console._clear_tooltip = lambda: cleared.append(True)
+
+        await pilot.resize_terminal(120, 40)
+        await pilot.pause()
+
+        assert cleared
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/UI/test_console_resize_reflow.py]: Fixed — the test now proves the real overlay: it shows the screen's `Tooltip` widget (`display=True`), resizes, and asserts `tooltip.display is False` afterwards, so it exercises the actual `_clear_tooltip` dismissal rather than a spy.

File: Tests/UI/test_console_resize_reflow.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # The panes are actually present (not the "panes gone" state) and the header
+    # is compacted at 30 rows.
+    assert cold["#console-transcript-surface"] is True
+    assert cold["#console-native-composer"] is True
+    assert cold["compact"] is True
+
+
+@pytest.mark.asyncio
+async def test_console_resize_dismisses_stale_tooltip():
+    """TASK-361 AC#2: a live resize dismisses a hover tooltip so a mounted
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/UI/test_console_resize_reflow.py]: Fixed — added `-> None` to both async test functions.

File: Tests/UI/test_console_resize_reflow.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        "-console-compact"
+    )
+    return layout
+
+
+@pytest.mark.asyncio
+async def test_console_live_resize_converges_to_cold_start_layout():
+    """TASK-361 AC#1: reflow after a live resize converges to the cold-start
+    layout at that size (panes present, header compacted), not the review's
+    rail-full-width / panes-gone divergence."""
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/UI/test_console_resize_reflow.py]: Reformatted — the new tests now use a one-line summary + blank line + Google-style body (this is a new file, so no existing convention to preserve).
